### PR TITLE
[PY-402][external] Archive Meta Item & ItemQuery functions

### PR DESCRIPTION
**Pull request by JBWilkie**
# Problem
Currently no Meta methods for archiving items

# Solution
Introduced functions to do this for Meta Items & ItemQuery objects

# Changelog
Meta methods to archive Items & ItemQuery objects

### CODE REVIEW

File: darwin/future/tests/meta/objects/test_itemmeta.py [hunk 1]
```diff
@@ -44,3 +44,26 @@ def test_delete(item: Item) -> None:
...
+                        "filters": {
+                            "item_ids": [str(item.id)],
+                            "dataset_ids": [dataset_id],
+                        }
+                    }
+                )
+            ],
+            json={},
+        )
+        item.archive()
```
Comment (owencjones) [inline darwin/future/tests/meta/objects/test_itemmeta.py:69]: Test the assert in the `archive` function works too.

File: darwin/future/tests/meta/queries/test_item.py [hunk 2]
```diff
@@ -70,3 +70,38 @@ def test_delete(
...
+                        "filters": {
+                            "item_ids": [str(item.id) for item in items],
+                            "dataset_ids": [dataset_id],
+                        }
+                    }
+                )
+            ],
+            json={},
+        )
+        item_query.archive()
```
Comment (owencjones) [inline darwin/future/tests/meta/queries/test_item.py:107]: Sad path opp here too.

**Review by owencjones — APPROVED** [review]
Some more sad path opportunities in these tests, but the code is sound and getting really consistent.  Good stuff.  I'll approve so you can make changes at discretion.

File: darwin/future/tests/meta/queries/test_item.py (hunk 2 above)
Comment (JBWilkie) [reply darwin/future/tests/meta/queries/test_item.py:107]: @owencjones Similar to my comment on the restore meta ticket and after discussion w/ Nathan, there is only one path this function can take due to no argument & no return

Will proceed with this test alone as error responses are tested by the underlying Core method, but let me know if you disagree

**Pull request merged**